### PR TITLE
Add expanding text box to title field in submission page

### LINKED ISSUE #1667: Manuscript title input should wrap/increase in height
Currently some authors are missing extraction errors in their manuscript titles. The assumption is that they miss it because it's not shown/obscured because the manuscript title input is only tall enough to show one line of text.

It should instead wrap/grow to show the entire manuscript title, up to a maximum height in case of extraction errors... perhaps 3 lines of text tall?

image

~The box should also allow rich text input: bold, italics, sub and superscript~

### PR DESCRIPTION
**Pull request by petereast**
#### Background

Adds and expanding text box to the SubmissionPage title field.

#### Any relevant tickets
closes #1667 

#### How has this been tested?
**Reviewers** ensure that these test requirements are also reviewed.

Please indicate the need for this change to be tested. If not please justify, if so then at what levels. 

- [x] Unit / Integration tests
- [ ] Browser tests
This part of the UI is already covered with browser tests, and still passes with the new component.
- [ ] End2end tests

#### UI Changes

- [ ] Has been reviewed against Designs
- [ ] Necessary updates to styleguide

### CODE REVIEW

File: packages/component-submission/client/components/ExpandingTextField.js [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+const TextArea = styled.textarea`
+  border: ${th('borderWidth')} ${th('borderStyle')} ${validationColor};
+
+  border-radius: ${th('borderRadius')};
+
+  font-family: inherit;
+  font-size: inherit;
+
+  padding: ${th('gridUnit')};
+  /* height: calc(${th('gridUnit')} * 6); */
```
Comment (erezmus) [inline packages/component-submission/client/components/ExpandingTextField.js:31]: remove commented line

**Review by erezmus — CHANGES_REQUESTED** [review]

**Comment by hdrury1** [comment]
I get an error when I try to access the review environment... do we need to update the AWS keys in the pipeline?

File: packages/component-submission/client/pages/SubmissionPage.js [hunk 2]
```diff
@@ -8,12 +8,20 @@ import {
 } from '@elifesciences/component-elife-ui/client/atoms'
 
 import SubjectAreaDropdown from '../components/SubjectAreaDropdown'
+import ExpandingTextField from '../components/ExpandingTextField'
 import OptionalSection from '../components/OptionalSection'
 
 const SubmissionPage = ({ values, setFieldValue, setFieldTouched }) => (
+  // TODO: turn this ValidatedField into a richTextBox
```
Comment (Cooryd) [inline packages/component-submission/client/pages/SubmissionPage.js:15]: remove comment if done

File: packages/component-submission/client/components/ExpandingTextField.js [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+    } = this.props
+
+    return (
+      <Root className={className} inline={inline}>
+        {label && <Label htmlFor={this.inputId}>{label}</Label>}
+        <TextArea
+          id={this.inputId}
+          readOnly={readonly}
+          ref={this.textAreaRef}
+          rows={this.state.lines || minRows}
```
Comment (Cooryd) [inline packages/component-submission/client/components/ExpandingTextField.js]: Is there a case where rows might be set to something less than `this.props.minRows`?

**Review by erezmus — APPROVED** [review]

**Review by will-byrne — APPROVED** [review]

**Review by Cooryd — APPROVED** [review]

**Pull request merged**